src/accelerate/accelerator.py

```diff
@@ -324,10 +324,13 @@ def __init__(
 
         if is_ipex_available():
             if ipex_plugin is None:  # init from env variables
-                ipex_plugin = IntelPyTorchExtensionPlugin()
+                ipex_plugin = (
+                    IntelPyTorchExtensionPlugin() if os.environ.get("IPEX_ENABLED", "true") == "true" else None
+                )
```

Comment:
Still not entirely sure I agree with this.

Is it possible to have `is_ipex_available` check if the device supports IPEX specifically, not just if it's installed?

Because we *should* pick up `ipex` by default if it's usable and available on the system, that is how all of accelerate's device configurations work. cc @abhilash1910 

(So basically if this PR is needed, we have other issues I think)

Reply:
If we want to do this, we should store a "disable IPEX" or something similar in the `PartialState` that gets set/read during `accelerate launch` and stored in the config file. 

In a follow-up PR I'll also adjust the env vars to act similar to Accelerate's

Reply:
@muellerzr , do you mean remove the env check to is is_ipex_available()?

Reply:
Yes, so this ipex plugin checking and creation should happen in the state, not in the `Accelerator` object. The `Accelerator` object should only handle the case of if the user wants to pass their own one in. And if a plugin is passed in and ipex is *not* enabled explicitly, we need to raise an error stating this and not just enable it magically on the backend. 

If needed, I can put in a PR today also changing this. Let me know if so, perfectly fine doing that.

Reply:
Hi, @muellerzr I get your point, do you think we should remove ipex_plugin input in Accelerator object and control ipex plugin in Accelerate state. the accelerate state only has ipex plugin if there's ipex installed and users do not disable it explicitly? I so, I could help to change like this and modify testcase as well.

Reply:
In general the Ipex plugin is there with just one flag... There is no need for a plugin for just one boolean flag, we can have the flag in the state :-)

Reply:
Agreed w/ Sylvains note, let's move forward with that @sywangyi, you can keep it in this PR if you'd like or do another one :) Just let us know when you're ready to review it

Reply:
I will update the PR. remove ipex plugin. only add enable flag in state